### PR TITLE
Add JSON support

### LINKED ISSUE #450: JSON support
Go client should support JSON communication with server.

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**
This PR adds JSON support.

Fixes #450.

### CODE REVIEW

File: serialization/internal/default_serializer.go [hunk 1]
```diff
@@ -413,3 +413,19 @@ func (*GobSerializer) Write(output serialization.DataOutput, i interface{}) erro
...
+	return JSONSerializationType
+}
+
+func (*HazelcastJSONSerializer) Read(input serialization.DataInput) (object interface{}, err error) {
+	obj, err := input.ReadByteArray()
+	return core.HazelcastJSON{JSONString: obj}, err
+}
+
+func (*HazelcastJSONSerializer) Write(output serialization.DataOutput, object interface{}) (err error) {
+	output.WriteByteArray(object.(core.HazelcastJSON).JSONString)
```
Comment (sancar) [inline serialization/internal/default_serializer.go]: We are serializing this as UTF on the java. this needs to be consistent.

**Comment by SaitTalhaNisanci** [comment]
verify

**Comment by Holmistr** [comment]
verify

**Comment by SaitTalhaNisanci** [comment]
verify

File: serialization/internal/default_serializer.go (hunk 1 above)
Comment (SaitTalhaNisanci) [reply serialization/internal/default_serializer.go]: @sancar  updated.

**Pull request merged**